pkg/user/spam_txclient/main.go

```diff
@@ -0,0 +1,194 @@
+package main
+
+import (
+	"context"
+	cryptorand "crypto/rand"
+	"errors"
+	"fmt"
+	"log"
+	"math"
+	"math/rand"
+	"sync/atomic"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/v6/app"
+	"github.com/celestiaorg/celestia-app/v6/app/encoding"
+	"github.com/celestiaorg/celestia-app/v6/pkg/user"
+	"github.com/celestiaorg/go-square/v3/share"
+	"github.com/cosmos/cosmos-sdk/client/grpc/cmtservice"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	mochaEndpoint   = "rpc-mocha.pops.one:9090"
+	blobSizeKB      = 300  // 300 KiB blobs
+	intervalMs      = 1000 // Submit every 1 second
+	testDurationSec = 120  // Run for 120 seconds
+)
+
+func main() {
+	log.Println("Setting up tx client and connecting to a mocha node")
+
+	// Global test context with configured timeout
+	ctx, cancel := context.WithTimeout(context.Background(), testDurationSec*time.Second)
+
+	txClient, grpcConn, _, err := NewMochaTxClient(ctx)
+	if err != nil {
+		log.Fatalf("failed to set up tx client: %v", err)
+	}
+
+	var (
+		txCounter            int64
+		successfulBroadcasts atomic.Int64
+		successfulConfirms   atomic.Int64
+		failedConfirms       atomic.Int64
+		failedBroadcasts     atomic.Int64
+		lastSuccess          atomic.Int64 // store UnixNano of last successful broadcast
+	)
+	// Initialize last success to now
+	lastSuccess.Store(time.Now().UnixNano())
+
+	g, ctx := errgroup.WithContext(ctx)
+
+	ticker := time.NewTicker(time.Duration(intervalMs) * time.Millisecond)
+
+	// Submission loop which breaks out if the text duration is exceeded
```

Comment:
what is the text duration? Maybe meant to be test duration?


```suggestion
	// Submission loop which breaks out if the test duration is exceeded
```